File: frontend/src/types/payment.ts
import type { OrderEntity } from "./orders";

export type PaymentMethod =
    | "CASH"
    | "CREDIT_CARD"
    | "DEBIT_CARD"
    | "BANK_TRANSFER"
    | "E_WALLET"
    | "QR_CODE";

export type PaymentStatus = "PENDING" | "SUCCESS" | "FAILED" | "REFUNDED";

export type PaymentEntity = {
    paymentId: number;
    order: OrderEntity;
    amount: number;
    method: PaymentMethod;
    transactionId: string | null;
    providerResponse: string | null;
    paidAt: string | null;
    createdAt: string;
    updatedAt: string;
    status: PaymentStatus;
};

export type Payment = PaymentEntity;

File: backend/src/main/java/com/dev/cinemasystem/entity/Payment.java
package com.dev.cinemasystem.entity;

import com.dev.cinemasystem.enums.PaymentStatus;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payment")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer paymentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    Order order;

    @Column(nullable = false, precision = 12, scale = 2)
    BigDecimal amount;

    String method;

    String bankCode;

    String bankTransactionNo;

    @Column(length = 200, unique = true)
    String transactionId;

    String infoTransaction;

    LocalDateTime paidAt;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    LocalDateTime updatedAt;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    PaymentStatus status = PaymentStatus.PENDING;
}
